test/e2e/soak_k8petstore.go

```diff
@@ -0,0 +1,231 @@
+				break T
+			}
+			if totalTransactions < 3 {
+				break T
+			}
+			ready = true
+		case <-tick:
+			//Pass if we've collected enough transactions
+			err, totalTransactions = readTransactions(c, ns)
+			if ready {
+				Expect(err).NotTo(HaveOccurred())
+			}
+			if totalTransactions > minExpected {
+				break T
+			}
+			Logf("%v == %v total petstore transactions stored into redis. ==", time.Now(), totalTransactions)
+		}
+	}
+
+	//Finally! We should have exceeded the min_expected num of transactions.
+	//If this fails, but there are transactions being created, we may need to recalibrate
+	//the min_expected value - or else - your cluster is broken/slow !
+	Ω(totalTransactions).Should(BeNumerically(">", minExpected))
+}
+
+var _ = Describe("K8PetStore (succesfull launching and transactions in the k8petstore application)", func() {
+
+	var namespace *api.Namespace = nil
+	var ns string = ""
+
+	//This must run on a machine which is in the kube-proxy ring.  Otherwise, the publicIP binding will fail.
+	//The IP Below ~ letter->number cipher for k8petstore (most likely it won't be bound by any other process)
+	var ip = "165.201.92.15"
+
+	//Used for calibrating expected transaction #s in the soak portion.
+	var minionCount int
+
+	BeforeEach(func() {
+		By("Creating a kubernetes client")
+		c, err := loadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Building a namespace api object")
+		namespace, err = createTestingNS("k8petstore-soak", c)
+		ns = namespace.Name
+		Expect(err).NotTo(HaveOccurred())
+
+		//Now get the # of minions and calibrate the test params to them...
+		minions, err := c.Nodes().List(labels.Everything(), fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+		minionCount = len(minions.Items)
+
+	})
+
+	AfterEach(func() {
+		c, err := loadClient()
+		Expect(err).NotTo(HaveOccurred())
+		By(fmt.Sprintf("Destroying namespace for this test %v", namespace.Name))
+		if err := c.Namespaces().Delete(namespace.Name); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+	})
+
+	//On a single node cluster, we expect about 10 transactions every second on average.
+	//admittedly, this is a very rough estimate given that the ETL is done in batches...
+	It(fmt.Sprintf("Should be FUNCTIONALLY able to successfully acquire 500 transactions per minion"), func() {
+		c, err := loadClient()
+		Expect(err).NotTo(HaveOccurred())
+		//max number of trial before k8petstore.sh dies.
+		var loadGenerators int = minionCount
+		var restServers int = minionCount
+
+		fmt.Printf("load generators / rest generators [ %v  /  %v ] ", loadGenerators, restServers)
+
+		//At least 500 transactions should be acquired within a short time period.
+		runK8petstore(ip, restServers, loadGenerators, c, ns, 500, 60*5)
+
+	})
+
+	//nThis test takes a few minutes... for simple CI setups testing pure functionality, filter it out.
+	PIt(fmt.Sprintf("should SCALE to support generating and storing several thousands of transactions per node"), func() {
```

Comment:
long term, that exact number may vary.  but in any case, ill update it to be precise for now.  
over time id like this to be user configurable and based on some cluster heuristics / metadata if that is possible.